infrastructure/application/utils/failureNotification.ts

```diff
@@ -0,0 +1,50 @@
+"use strict";
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+
+/**
+ * Send a Slack notification when a Fargate service rolls back it's deployment
+ * Any initial Pulumi errors already handle this via notifications from GHA
+ * This will catch instances where the container has successfully deployed, but may be unhealthy and is caught by a circuit breaker
+ */
+export const setupFailureNotificationForDeployments = (
+  serviceName: string,
+  cluster: awsx.ecs.Cluster,
+  service: awsx.ecs.FargateService
+) => {
+  const config = new pulumi.Config();
+
+  const topic = new aws.sns.Topic(`${serviceName}-deployment-alerts`, {
+    name: `${serviceName}-deployment-alerts`,
+  });
+
+  new aws.sns.TopicSubscription(
+    `${serviceName}-slack-alert`,
+    {
+      topic: topic.arn,
+      protocol: "https",
+      endpoint: config.requireSecret("slack-internal-errors-webhook"),
+    }
+  );
+
+  new aws.cloudwatch.MetricAlarm(
+    `${serviceName}-deployment-alarm-notification`,
+    {
+      alarmActions: [topic.arn],
+      comparisonOperator: "GreaterThanThreshold",
+      evaluationPeriods: 1,
+      metricName: "ECSServiceFailedTaskCount",
```

Comment:
Only thing I'm a bit unsure about is this metric? I can't see it referenced in AWS docs, e.g. [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/available-metrics.html) or [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-metrics-ECS.html). Of course, that absolutely doesn't mean that it doesn't exist 😅

Reply:
[This article](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/deployment-alarm-failure.html) suggests using metrics on the ALB 🤔

Reply:
After a bit of poking around I think `UnhealthyTaskCount` might be the right metric? Based on the fact that a rollback should occur only in the case that the new task cannot be booted to a healthy state, and is therefore labelled as unhealthy

Reply:
Thanks - struggled with this one as I couldn't find an appropriate metric via the console and came across this somewhere when looking for a solution.

I've now updated to `UnhealthyTaskCount` - seems more reasonable!

Reply:
@DafyddLlyr awesome, sounds good! Let's test this and see how it goes. It may be we need to reduce the period over which the unhealthy task count needs to be sustained (since it may be killed before 5 mins are up).